R/tm_g_heat_bygrade.R

```diff
@@ -336,24 +343,27 @@ srv_g_heatmap_bygrade <- function(id,
     })
 
     plt <- reactive({
-      validate(need(input$id_var, "Please select a ID variable."))
-      validate(need(input$visit_var, "Please select a visit variable."))
-      validate(need(input$ongo_var, "Please select a Study Ongoing Status variable."))
-      validate(need(input$heat_var, "Please select a heat variable."))
-      validate(need(length(input$anno_var) <= 2, "Please include no more than 2 annotation variables"))
+      iv_len <- shinyvalidate::InputValidator$new()
+      anno_var <- input$anno_var
```

Comment:
this is defined but not actually used, either remove or replace the downstream calls of `input$anno_var`.

Reply:
I did it like this, because I was not able to make this:
```
iv_len$add_rule("input$anno_var", function(x) if (length(x) > 2) "Please include no more than 2 annotation variables.")
```

work, it only accepted "anno_var"

Reply:
calls changed